### PR TITLE
Jetstream + RayServe deployment for interleave mode

### PR DESCRIPTION
**Pull request by richardsliu**

### CODE REVIEW

**Review by FanhaiLu1 — APPROVED** [review]
LGTM, thanks for adding this feature!

File: kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+        serviceAccountName: ray-ksa
+        containers:
+          - volumeMounts:
+            - name: gcs-fuse-checkpoint
+              mountPath: /llama
+              readOnly: true
+            - mountPath: /tmp/ray
+              name: ray-logs
+            name: ray-head
+            image: gcr.io/tpu-vm-gke-testing/ricliu-jetstream:20240709
```
Comment (andrewsykim) [inline kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml:25]: This image is not publically available and references an internal project, can we host it on a public registry or provide the Dockerfile so users can build it themselves?

File: kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+                name: grpc
+        volumes:
+          - emptyDir: {}
+            name: ray-logs
+          - name: gcs-fuse-checkpoint
+            csi:
+              driver: gcsfuse.csi.storage.gke.io
+              readOnly: true
+              volumeAttributes:
+                bucketName: ricliu-llama2-70b-chat
```
Comment (andrewsykim) [inline kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml:72]: This bucket is not publically available, can we host it in a public bucket or provide instructions to push the model weights to a bucket?

Comment (andrewsykim) [reply kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml:72]: cc @ryanaoleary

File: kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml (hunk 1 above)
Comment (andrewsykim) [reply kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml:25]: cc @ryanaoleary

File: README.md [hunk 3]
```diff
@@ -199,6 +199,92 @@ python benchmarks/benchmark_serving.py --tokenizer $tokenizer_path --num-prompts
...
+### Start a Ray Serve deployment
+
+Single-host (Llama2 7B):
+
+```bash
+export RAY_ADDRESS=http://localhost:8265
+
+kubectl port-forward svc/example-cluster-kuberay-head-svc 8265:8265 &
+
+ray job submit --runtime-env-json='{"working_dir": "."}' -- python run_ray_serve_interleave.py  --tpu_chips=4 --num_hosts=1 --size=7b --model_name=llama-2 --batch_size=32 --max_cache_length=2048 --tokenizer_path=/llama/tokenizer.model --checkpoint_path=/llama/ckpt --quantize_weights=True --quantize_type="int8_per_channel" --quantize_kv_cache=True --sharding_config="default_shardings/llama.yaml"
```
Comment (andrewsykim) [inline README.md:233]: I don't think `ray job submit` was intended to be used with Ray Serve in this way. If you run it like this, the Ray Serve application will be treated as a Ray job and not survive a restart. Could we provide an example that uses the `serve run` CLI or the KubeRay RayService? cc @ryanaoleary

File: kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml (hunk 1 above)
Comment (andrewsykim) [reply kuberay/manifests/ray-cluster.tpu-v4-multihost.yaml:25]: Ah I see the Dockerfile now. I suggest not referencing the private image though if possible.

**Pull request merged**